iothub_client/src/iothubtransport_amqp_telemetry_messenger.c

```diff
@@ -623,11 +627,11 @@ static AMQP_VALUE create_uamqp_disposition_result_from(TELEMETRY_MESSENGER_DISPO
 static AMQP_VALUE on_message_received_internal_callback(const void* context, MESSAGE_HANDLE message)
 {
     AMQP_VALUE result;
-    int api_call_result;
     IOTHUB_MESSAGE_HANDLE iothub_message;
 
     // Codes_SRS_IOTHUBTRANSPORT_AMQP_MESSENGER_09_121: [An IOTHUB_MESSAGE_HANDLE shall be obtained from MESSAGE_HANDLE using message_create_IoTHubMessage_from_uamqp_message()]
-    if ((api_call_result = message_create_IoTHubMessage_from_uamqp_message(message, &iothub_message)) != RESULT_OK)
+    int api_call_result = message_create_IoTHubMessage_from_uamqp_message(message, &iothub_message);
```

Comment:
So I assume the reason the variable was added in the first place was for debuggability. Otherwise, you have to be a gdb/windbg god to know how to get the return value out of the registers or stack frame (where is it stored in compilers these days? is it the same across all supported compilers?) Given this is intended to be a cross platform library, I'd think having the return value stored in a easy to read place would be a benefit.